### PR TITLE
Move simple simulator and model from hicat-package2 to catkit2

### LINKED ISSUE #187: Move simple simulator and optical model into catkit2
Example code on how to use the optical model and simulator are currently contained in HiCAT-specific repository. There is a broader interest in these example scripts, so they should be moved to the catkit2 repo.

### PR DESCRIPTION
**Pull request by ivalaginja**
Fixes #187.

Pair PR with https://github.com/spacetelescope/hicat-package2/pull/587.

Needs testing and converting the added test into a real test that pytest understands.

### CODE REVIEW

**Comment by ivalaginja** [comment]
This is ready for review, although I don't really know how to make the test make more sense, and whether it's worth it.

File: tests/test_optical_model.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    pupil_grid = model.pupil_grid
+
+    wf_in = hcipy.Wavefront(pupil_grid.ones())
+    model.set_wavefronts('pre_pupil', wf_in)
+    
+    test_wf = model.get_wavefronts('science_camera')[0]
+    size1 = test_wf.intensity.shaped.shape[0]
+    size2 = test_wf.intensity.shaped.shape[1]
+
+    stream = DataStream.create('images', 'mod', 'float64', [size1, size2], 20)
```
Comment (ehpor) [inline tests/test_optical_model.py]: I'm not sure why the data stream is needed here. This is a leftover from very early on, when I wanted to test an early version of the camera viewer UI.

**Comment by ehpor** [comment]
> This is ready for review, although I don't really know how to make the test make more sense, and whether it's worth it.

The test looks fine to me. Basic functionality test and checks that the code doesn't crash.

File: catkit2/services/simple_simulator/simple_simulator.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    def update_atmosphere(self):
+        self.log.info('Updating atmosphere.')
+
+        t = self.time.get()[0]
+
+        self.model.atmosphere.t = t
+        self.model.purge_plane('pre_coro')
+
+        self.add_callback(t + 0.01, self.update_atmosphere)
```
Comment (raphaelpclt) [inline catkit2/services/simple_simulator/simple_simulator.py:40]: Does this recursive call work? The `update_atmosphere` puts itself in the callback?

Comment (raphaelpclt) [reply catkit2/services/simple_simulator/simple_simulator.py:40]: Shouldn't be a callback function be defined here actually?

File: catkit2/simulator/simple_optical_model.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        outer_scale = 20
+        velocity = 0.3
+
+        Cn_squared = hcipy.Cn_squared_from_fried_parameter(r0, 1)
+        return hcipy.InfiniteAtmosphericLayer(self.pupil_grid, Cn_squared, outer_scale, velocity)
+
+    @property
+    @with_cached_result
+    def coro(self):
+        return hcipy.PerfectCoronagraph(hcipy.make_hicat_aperture(True)(self.pupil_grid))
```
Comment (raphaelpclt) [inline catkit2/simulator/simple_optical_model.py:64]: Do you want to keep hicat aperture here?

File: tests/test_optical_model.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    wf_in = hcipy.Wavefront(pupil_grid.ones())
+    model.set_wavefronts('pre_pupil', wf_in)
+
+    for i in range(5):
+        model.atmosphere.t += 0.01
+        model.purge_plane('pre_coro')
+
+        wf_out = model.get_wavefronts('science_camera')[0]
+
+        assert np.sum(wf_out.intensity) > 0
```
Comment (raphaelpclt) [inline tests/test_optical_model.py:21]: I am dubious about this test - why doing 5 iterations if it's to test the same thing every time? Why not test that the `wf_out` has changed? ^^
Also I understand it's a low-key test, so I don't think that's very important.

**Review by raphaelpclt — CHANGES_REQUESTED** [review]
Looks good overall. 
Just a few small questions, nothing serious

File: catkit2/simulator/simple_optical_model.py (hunk 3 above)
Comment (ivalaginja) [reply catkit2/simulator/simple_optical_model.py:64]: Sure, why not. It's one of the apertures from `hcipy` so there's no harm.

File: tests/test_optical_model.py (hunk 4 above)
Comment (ivalaginja) [reply tests/test_optical_model.py:21]: You're totally right, but the only thing to test here is that it runs at all. The assert is rather random 🤷

File: catkit2/services/simple_simulator/simple_simulator.py (hunk 2 above)
Comment (ivalaginja) [reply catkit2/services/simple_simulator/simple_simulator.py:40]: Yes, it recursively adds a callback function that will be executed in the next iteration. In that next iteration, the execution happens and the callback creation for the following iteration happens and so on.

**Comment by ivalaginja** [comment]
@raphaelpclt all fair questions, I hope I answered them sufficiently, ready for a new review. Let me know if you'd need more.

**Review by raphaelpclt — APPROVED** [review]
Looks good

**Pull request merged**